index.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs')
+const core = require('@actions/core')
+const tc = require('@actions/tool-cache')
+const io = require('@actions/io')
+const exec = require('@actions/exec')
+
+async function installLinux (version) {
+  const dmenvDlPath = await tc.downloadTool(`https://github.com/TankerHQ/dmenv/releases/download/v${version}/dmenv-linux`)
+  await installToLocalBin(dmenvDlPath)
+
+  // Don't trust debian-based distros with Python packaging
+  //   - install pip with get-pip.py instead of the debian python3-pip package
+  //   - install virtualenv with python3 -m pip instead of the python3-venv or python3-virtualenv debian package
+  //   - tell dmenv to *not* use `python -m venv` when creating virtual environments
+  const getPipPath = await tc.downloadTool('https://bootstrap.pypa.io/get-pip.py')
+  await exec.exec('python', [getPipPath, '--user'])
+  await exec.exec('python', ['-m', 'pip', 'install', 'virtualenv', '--user'])
+  core.exportVariable('DMENV_NO_VENV_STDLIB', '1')
+}
+
+async function installDarwin (version) {
+  const dmenvDlPath = await tc.downloadTool(`https://github.com/TankerHQ/dmenv/releases/download/v${version}/dmenv-osx`)
+  await installToLocalBin(dmenvDlPath)
+}
+
+async function installWindows (version) {
+  const dmenvDlPath = await tc.downloadTool(`https://github.com/TankerHQ/dmenv/releases/download/v${version}/dmenv-windows.exe`)
+
+  const installPath = 'c:\\dmenv'
+  await io.mkdirP('c:\\dmenv')
+  await io.cp(dmenvDlPath, `${installPath}\\dmenv.exe`)
+
+  core.addPath(installPath)
+}
+
+async function installToLocalBin (dlPath) {
+  const binariesPath = `${process.env.HOME}/.local/bin`
+  await io.mkdirP(binariesPath)
+  const dmenvBinPath = `${binariesPath}/dmenv`
+  await io.cp(dlPath, dmenvBinPath)
+  await fs.chmod(dmenvBinPath, 0o775, (err) => {
```

Comment:
This await does not await anything.

Either use `fs.chmodSync` without awaiting, or wrap the callback-based `fs.chmod` with a Promise:

```js
await new Promise((resolve, reject) => {
  fs.chmod(dmenvBinPath, 0o775, (err) => err ? reject(err) : resolve());
});
```